src/revpi-factory-reset

```diff
@@ -8,14 +8,24 @@
 # reset dtoverlay, password of user pi, machine-id, hostname, FQDN and
 # MAC address of a RevPi to factory defaults
 
+. /usr/share/revpi/revpi-functions
+
 ovl="$1"
 ser="$2"
 mac="$3"
 mac="${mac,,}"		# downcase
 mac="${mac//-/}"	# remove dashes
 mac="${mac//:/}"	# remove colons
 
-full_devicetype_list="compact|connect|core|flat|connect-se"	# all suported devices
+if cm1-detection; then
+	echo "Compute Module 1 has been detected, core-cm1-overlay will be used"
+	echo "automatically."
+	ovl="core-cm1"
+	full_devicetype_list="core-cm1"	# only suported device
+else
+	full_devicetype_list="compact|connect|core|flat|connect-se"	# all suported devices
+fi
+
```

Comment:
The requirement was to not give the user any other option than CM1 if one is detected. Perhaps this should be taken through Nico first.

Reply:
Hmmm, why don't you put the core-cm1 just in the full_..._list, and ignore the autodetect in this case?

Reply:
It still must be possible to chose from all available device types in the backend script (even on Core with CM1). If a Core was choosen, the detection must be triggered and if positive the respective dtoverlay must be chossen (core-cm1). Also an information message must be presented to the user (like: "NOTE: Core with CM1 was detected. Choosing revpi-core-cm1 instead of revpi-core as device tree overlay ...")

In the frontend (the whiptail stuff) only the Core CM1 should be available (as implemented by you).

Reply:
With this change I receive an error that did not occur before on CM1
```
* Failed to apply overlay '0_revpi-core-cm1' (kernel)
```
This leads to an error at the end of revpi-factory-reset.sh, as well. My search indicated that the source is `/opt/vc/bin/dtoverlay` but I still try to figure out why this happened with the suggestion and not without.

Reply:
this error message should be caused by repeated node labels.

Reply:
That this message somehow created fallout in the calling script was interesting but I have fixed this now.